providers/qedr/qelr_verbs.c

```diff
@@ -922,14 +922,17 @@ static inline void qelr_edpm_set_inv_imm(struct qelr_qp *qp,
 					 struct qelr_edpm *edpm,
 					 uint32_t inv_key_or_imm_data)
 {
+	uint32_t data;
+
 	if (!edpm->is_edpm)
 		return;
 
-	memcpy(&edpm->dpm_payload[edpm->dpm_payload_offset],
-	       &inv_key_or_imm_data, sizeof(inv_key_or_imm_data));
+	data = htobe32(inv_key_or_imm_data);
+
```

Comment:
The byte swapping here is nutzo..

```
                rwqe->imm_data = htobe32(htole32(wr->imm_data));
                qelr_edpm_set_inv_imm(qp, &edpm, rwqe->imm_data);
+	data = htobe32(inv_key_or_imm_data);
￼```

Can you fix this so it makes sense - and would pass sparse?

wr->imm_data is defined to be be32, so you probably just need to memcpy that and remove all the swaps completely.